### PR TITLE
Update e621dl.py

### PR DESCRIPTION
**Pull request by RedFoxNightfox**
added error catching to retry on 10054 and other network errors.

added logging to log file.

### CODE REVIEW

File: e621dl.py [hunk 1]
```diff
@@ -2,10 +2,16 @@
 
 import os
 import httpx
+import logging
 from lib import constants
 from lib import local
 from lib import remote
 
+LOG_FILENAME = 'ErrorLog.txt'
```
Comment (Wulfre) [inline e621dl.py:10]: Move to constants module.

File: e621dl.py [hunk 2]
```diff
@@ -2,10 +2,16 @@
...
+logging.basicConfig(filename=LOG_FILENAME, level=logging.ERROR, format="%(asctime)s %(message)s")
+CheckLogs = 'FALSE'
```
Comment (Wulfre) [inline e621dl.py:12]: Use boolean instead of string, follow naming convention for variables.
`checkLogs = False`

File: e621dl.py [hunk 3]
```diff
@@ -38,8 +44,21 @@
...
-                posts = remote.get_posts(client, ' '.join(search['tags']), search['start_date'], last_id)
                 
+                while True:
+                    try:
+                        posts = remote.get_posts(client, ' '.join(search['tags']), search['start_date'], last_id)
+                        break
+                    except TimeoutError:
+                        print("[✗] TimeoutError. retrying.")
+                        logging.exception('TimeoutError')
+                        CheckLogs = 'TRUE'
```
Comment (Wulfre) [inline e621dl.py:55]: Use boolean instead of string, follow naming convention for variables.

File: e621dl.py [hunk 4]
```diff
@@ -38,8 +44,21 @@
...
+                    except httpx._exceptions.NetworkError:
+                        print("[✗] NetworkError. retrying.")
+                        logging.exception('NetworkError')
+                        CheckLogs = 'TRUE'
```
Comment (Wulfre) [inline e621dl.py:59]: Use boolean instead of string, follow naming convention for variables.

File: e621dl.py [hunk 5]
```diff
@@ -61,4 +80,7 @@
                     break
 
     print('[i] All searches complete.')
+    if CheckLogs == 'TRUE':
```
Comment (Wulfre) [inline e621dl.py:83]: Use boolean instead of string, follow naming convention for variables.

**Review by Wulfre — CHANGES_REQUESTED** [review]
Sorry for missing your original issue. I've been busy with other projects and didn't see the email for it.

I added some quick reviews to your request that are all semantic, the functionality looks fine. I can merge your pull and make the changes myself, but I wanted to see if you'd like to do them first so that you get full credit for the fix.

**Comment by RedFoxNightfox** [comment]
Made the changes requested.

**Comment by Wulfre** [comment]
Thanks for that. I know this hasn't been updated in a long time, but it makes it easier to go back read in the future if everything is consistent.

**Pull request merged**